### DISCUSSION TITLE
Electric Frontdoor Opener in combination with Contact Sensor (Reed) with a Shelly1

### DISCUSSION THREAD

**Discussion by lickonline** [RavenSystem/esp-homekit-devices#1541] [MEPLHAA Scripts Collection]
• Shelly 1
• -
• HAA v6.10.0
• 1 Electric Frontdoor Opener (GPIO4) with Button to unlock (GPIO5)    and   Door Contact Sensor (GPIO1) (additional Reed Switch)
• {"a":[{"t":4,"i":2.5,"0":{"r":[{"g":4,"v":1}]},"1":{"r":[{"g":4}]},"b":[{"g":5}],"s":1},{"t":5,"0":{"r":[{"g":3,"v":0}]},"1":{"r":[{"g":3,"v":1}]},"f0":[{"g":1,"t":0}],"f1":[{"g":1,"t":1}],"f2":[{"g":1,"t":1}],"f3":[{"g":1,"t":0}]}]}
• -
• -
_____________________________________________________________________________
 
update:
after a power blackout / a (re)boot of the shelly I had a permanent activated door opener (no idea why)
but here is the new working json, with out this problem:

• Shelly 1
• -
• HAA v7.1.2
• 1 Electric Frontdoor Opener (GPIO4) and Door Contact Sensor (GPIO3) State Input (GPIO1) (Reed Switch needed)
• {"a":[{"0":{"r":[{"g":4,"v":1}]},"1":{"r":[{"g":4}]},"t":4,"i":1.5},{"0":{"r":[{"g":1,"v":0}]},"1":{"r":[{"g":1,"v":1}]},"t":5,"f0":[{"g":3,"t":0}],"f1":[{"g":3,"t":1}],"f2":[{"g":3,"t":1}],"f3":[{"g":3,"t":0}]}]}
• -
• -

**Comment by ndv27** [comment]
@lickonline, thanks for your config!
Can you send your wire diagram? What is power voltage for Shelly1? My concern is about high AC voltage (230V) on GND. I suppose that external relay is needed to close GPIO1 and GND and to isolate it from high voltage.

**Comment by lickonline** [comment]
![wire diagram](https://user-images.githubusercontent.com/46582029/147421315-592c7fcf-77d7-4585-b614-91d3a0fec2d6.jpg)
